examples/docs-examples/examples/composing-tasks/batch-endstream-chunks.mjs

```diff
@@ -6,17 +6,21 @@ import { TaskExecutor } from "@golem-sdk/golem-js";
     yagnaOptions: { apiKey: "try_golem" },
   });
 
-  const result = await executor.run(async (ctx) => {
-    const res = await ctx
-      .beginBatch()
-      .uploadFile("./worker.mjs", "/golem/input/worker.mjs")
-      .run("node /golem/input/worker.mjs > /golem/input/output.txt")
-      .run("cat /golem/input/output.txt")
-      .downloadFile("/golem/input/output.txt", "./output.txt")
-      .endStream();
+  try {
+    const result = await executor.run(async (ctx) => {
+      const res = await ctx
+        .beginBatch()
+        .uploadFile("./worker.mjs", "/golem/input/worker.mjs")
+        .run("node /golem/input/worker.mjs > /golem/input/output.txt")
+        .run("cat /golem/input/output.txt")
+        .downloadFile("/golem/input/output.txt", "./output.txt")
+        .endStream();
 
-    res.on("data", (data) => (data.index == 2 ? console.log(data.stdout) : ""));
-    res.on("error", (error) => console.error(error));
-    res.on("close", () => executor.end());
```

Comment:
This sill requires fixing. We want to wait for the stream to close, before we `end/shutdown` the executor.

Reply:
Please check the updated code. Is that what you meant?

Reply:
Not really :) You did it already in the other place by:

```ts
res.once("close", resolve);
```

Reply:
@grisha87 I must admit i'm not sure what to do from here. I am not sure I understand what you want me to do.